core/org.wso2.carbon.user.core/src/main/java/org/wso2/carbon/user/core/ldap/ReadOnlyLDAPUserStoreManager.java

```diff
@@ -482,23 +482,26 @@ public boolean doAuthenticate(String userName, Object credential) throws UserSto
                 }
             } else {
                 name = getNameInSpaceForUserName(userName);
-                try {
-                    if (name != null) {
-                        if (debug) {
-                            log.debug("Authenticating with " + name);
+                // if it is the same user DN found in the cache no need of futher authentication required.
+                if (!failedUserDN.equalsIgnoreCase(name)) {
```

Comment:
There is a possibility that 'failedUserDN' can be null. Please check the reference